src/actors/builtin/storage_miner/storage_miner_actor_code.go

```diff
@@ -23,11 +24,43 @@ import (
 // Actor methods
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StorageMinerActorCode_I) GetWorkerVRFKey(rt Runtime) filcrypto.VRFPublicKey {
+func (a *StorageMinerActorCode_I) StageWorkerKeyChange(rt Runtime, key filcrypto.VRFPublicKey) (err error) {
 	h, st := a.State(rt)
-	ret := st.Info().WorkerVRFKey()
+	rt.ValidateImmediateCallerIs(st.Info().Owner())
+	util.IMPL_TODO() // verify public key syntax
+
+	keyAddr, err := addr.Address_Make_Key(node_base.NETWORK, addr.KeyHash(key))
+	if err != nil {
+		Release(rt, h, st)
+		return err
+	}
+
+	keyChange := &MinerKeyChange_I{
+		NewKey_:      keyAddr,
+		EffectiveAt_: rt.CurrEpoch() + node_base.MINER_KEY_CHANGE_FREEZE,
+	}
+
+	// note that this may replace another pending key change
+	st.Info()._updateMinerPendingKeyChange(keyChange)
 	Release(rt, h, st)
-	return ret
+	return nil
+}
+
+func (a *StorageMinerActorCode_I) CommitWorkerKeyChange(rt Runtime) (err error) {
+	h, st := a.State(rt)
```

Comment:
Should we verify the caller here @anorth ? I think not

Reply:
Probably yes (as the owner)

Reply:
checking it's `addr.CronActorAddr` given https://github.com/filecoin-project/specs/pull/776#discussion_r364480899